packages/tfchain_client/lib/src/contracts.dart

```diff
@@ -0,0 +1,153 @@
+import 'package:tfchain_client/generated/dev/types/pallet_smart_contract/types/contract.dart';
+import 'package:tfchain_client/generated/dev/types/pallet_smart_contract/types/contract_lock.dart';
+import 'package:tfchain_client/generated/dev/types/pallet_smart_contract/types/service_contract.dart';
+import 'package:tfchain_client/tfchain_client.dart';
+
+const twoWeeks = 1209600000;
+
+class QueryContracts {
+  final QueryClient client;
+  QueryContracts(this.client);
+
+  Future<Contract?> get({required BigInt contractId}) async {
+    final res =
+        await client.api.query.smartContractModule.contracts(contractId);
+    return res;
+  }
+
+  Future<BigInt?> getContractIdByActiveRentForNode(
+      {required int nodeId}) async {
+    final res = await client.api.query.smartContractModule
+        .activeRentContractForNode(nodeId);
+    return res;
+  }
+
+  Future<List<int>> getActiveContracts({required int nodeId}) async {
+    final res =
+        await client.api.query.smartContractModule.activeNodeContracts(nodeId);
+    return res;
+  }
+
+  Future<BigInt?> getContractIdByName({required String name}) async {
+    final res = await client.api.query.smartContractModule
+        .contractIDByNameRegistration(name.codeUnits);
+    return res;
+  }
+
+  Future<BigInt?> getContractIdByNodeIdAndHash(
+      {required int nodeId, required List<int> hash}) async {
+    final res = await client.api.query.smartContractModule
+        .contractIDByNodeIDAndHash(nodeId, hash);
+    return res;
+  }
+
+  Future<ContractLock?> contractLock({required BigInt id}) async {
+    final res = await client.api.query.smartContractModule.contractLock(id);
+    return res;
+  }
+
+  Future<num> getDeletionTime({required BigInt id}) async {
+    final contract = await get(contractId: id);
+    if (contract != null && contract.state.toJson()["Created"] == null) {
+      return 0;
+    } else if (contract == null) {
+      return 0;
```

Comment:
we can combine those in
```suggestion
    if (contract == null || contract != null && contract.state.toJson()["Created"] == null) {
      return 0;
```